frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts

```diff
@@ -62,6 +70,46 @@ export const windowsToolsStorage: VMWizardStorage = {
   }).asResource(),
 };
 
+export const getBaseImageStorage = (
+  storageClassConfigMap: ConfigMapKind,
+  pvcName,
+  pvcNamespace,
+) => {
+  const dataVolumeName = generateDataVolumeName(DUMMY_VM_NAME, ROOT_DISK_NAME);
+
+  return {
+    type: VMWizardStorageType.PROVISION_SOURCE_DISK,
+    disk: DiskWrapper.initializeFromSimpleData({
+      name: ROOT_DISK_NAME,
+      type: DiskType.DISK,
+      bus: DiskBus.VIRTIO,
+      bootOrder: 1,
+    }).asResource(),
+    volume: VolumeWrapper.initializeFromSimpleData({
+      name: ROOT_DISK_NAME,
+      type: VolumeType.DATA_VOLUME,
+      typeData: { name: dataVolumeName },
+    }).asResource(),
+    dataVolume: new DataVolumeWrapper()
+      .init({
+        name: dataVolumeName,
+        size: 5,
+        unit: BinaryUnit.Gi,
+      })
+      .setType(DataVolumeSourceType.PVC)
+      .setVolumeMode(getDefaultSCVolumeMode(storageClassConfigMap))
+      .setAccessModes(getDefaultSCAccessModes(storageClassConfigMap))
+      .setPesistentVolumeClaimNamespace(pvcNamespace)
```

Comment:
these two should be set through setType 
- this method ensures consistency of type information.
- can you please also remove these two methods so they don't misguide people?

Reply:
done